### PR TITLE
feat(lsp): Add client command support to codelens

### PR DESCRIPTION
**Pull request by mfussenegger**
Also adds a check against the server capabilities to fix
https://github.com/neovim/neovim/issues/15183

### CODE REVIEW

File: runtime/lua/vim/lsp/codelens.lua [hunk 1]
```diff
@@ -31,10 +31,24 @@ local function execute_lens(lens, bufnr, client_id)
...
+  end
   -- Need to use the client that returned the lens → must not use buf_request
   local client = vim.lsp.get_client_by_id(client_id)
   assert(client, 'Client is required to execute lens, client_id=' .. client_id)
-  client.request('workspace/executeCommand', lens.command, function(...)
+  local command_provider = client.server_capabilities.executeCommandProvider
+  local commands = type(command_provider) == 'table' and command_provider.commands or {}
+  if not vim.tbl_contains(commands, command.command) then
+    vim.notify(string.format(
+      "Language server doesn't support command `%s`. This command may require a client extension", command.command),
```
Comment (mjlbach) [inline runtime/lua/vim/lsp/codelens.lua:47]: how do we feel about contractions? Also missing a terminating period.

Comment (mfussenegger) [reply runtime/lua/vim/lsp/codelens.lua:47]: Not sure? Happy to rephrase it if you've a suggestion.

Comment (mjlbach) [reply runtime/lua/vim/lsp/codelens.lua:47]: I was only suggesting "does not" and adding a period at the end

**Review by mjlbach — APPROVED** [review]

File: runtime/lua/vim/lsp/codelens.lua (hunk 1 above)
Comment (mfussenegger) [reply runtime/lua/vim/lsp/codelens.lua:47]: Oh gotcha. Changed it.

File: runtime/lua/vim/lsp/codelens.lua [hunk 2]
```diff
@@ -31,10 +31,24 @@ local function execute_lens(lens, bufnr, client_id)
...
+  local fn = vim.lsp.commands[command.command]
+  if fn then
+    fn(command, { bufnr = bufnr, client_id = client_id })
+    return
+  end
   -- Need to use the client that returned the lens → must not use buf_request
   local client = vim.lsp.get_client_by_id(client_id)
   assert(client, 'Client is required to execute lens, client_id=' .. client_id)
-  client.request('workspace/executeCommand', lens.command, function(...)
+  local command_provider = client.server_capabilities.executeCommandProvider
```
Comment (mjlbach) [inline runtime/lua/vim/lsp/codelens.lua:43]: Also I still dislike this pattern of sanitizing capabilities and think we should change it but it's consistent with everything else so not important.

**Pull request merged**